src/WinRT.Runtime/ComWrappersSupport.net5.cs

```diff
@@ -106,10 +106,13 @@ private static T CreateRcwForComObject<T>(IntPtr ptr, bool tryUseCache)
                 winrtObj.Resurrect();
             }
 
+            if (rcw.GetType().FullName.StartsWith("ABI.System.Nullable`1") || rcw.GetType().FullName.StartsWith("ABI.Windows.Foundation.IReferenceArray`1"))

```

Comment:
Can we somehow do pattern matching to handle this.  This right now I feel like will have some additional cost for each object creation.

Reply:
does adding ABI.System.Nullable<T> nullable => nullable.Value work?

Reply:
ABI.System.Nullable<T> nullable => nullable.Value does not seem to work when T is of object type. Nullable.Value requires the exact type of generic parameter.

Reply:
I wonder if it makes sense to create a non generic version of ABI.System.Nullable that we wrap the object return around and we detect that here and call .Value on it which returns it as an object and we cast it.  Thoughts?

Reply:
It could also just really be a new wrapper around the .Value object technically.